plotly/tools.py

```diff
@@ -1296,3 +1304,686 @@ def return_figure_from_figure_or_data(figure_or_data, validate_figure):
+    @staticmethod
+    def validate_streamline(x, y, u, v,
+                            density, angle,
+                            arrow_scale, **kwargs):
+        """
+        Validates the user-input arguments,
+
+        Specifically, that density and arrow_scale are positive
+        and that x and y are both evenly spaced.
+
+        :raises: (ImportError) If numpy is not imported.
+        :raises: (ValueError) If scale or arrow_scale is < 1.
+        :raises: (ValueError) If density or arrow_scale is < 1.
+        :raises: (PlotlyError) If x and y are not evenly spaced lists or
+            1D arrays.
+        :raises: (PlotlyError) If u and v are not the same shape.
+        """
+
+        if _numpy_imported is False:
+            raise ImportError("TraceFactor.create_streamline requires numpy.")
+        if arrow_scale <= 0:
+            raise ValueError("arrow_scale must be > 0")
+        if density <= 0:
+            raise ValueError("density must be > 0")
+        for index in range(len(x) - 1):
+            if ((x[index + 1] - x[index]) -
+               (x[1] - x[0])) > .0001:
+                raise exceptions.PlotlyError("x must be a 1 dimensional, "
+                                             "evenly spaced array")
+        for index in range(len(y) - 1):
+            if ((y[index + 1] - y[index]) -
+               (y[1] - y[0])) > .0001:
+                raise exceptions.PlotlyError("y must be a 1 dimensional, "
+                                             "evenly spaced array")
+        if len(u) != len(v):
+            raise exceptions.PlotlyError("u and v should both be 2d "
+                                         "arrays with the same "
+                                         "dimensions")
+
+    @staticmethod
+    def create_quiver(x, y, u, v, scale=.1, arrow_scale=.3,
+                      angle=math.pi / 9, **kwargs):
+        """
+        Returns data for a quiver plot.
+
+        :param (list|ndarray) x: x coordinates of the arrow locations
+        :param (list|ndarray) y: y coordinates of the arrow locations
+        :param (list|ndarray) u: x components of the arrow vectors
+        :param (list|ndarray) v: y components of the arrow vectors
+        :param (float in [0,1]) scale: scales size of the arrows(ideally to
+            avoid overlap). Default = .1
+        :param (float in [0,1]) arrow_scale: value multiplied to length of barb
+            to get length of arrowhead. Default = .3
+        :param (angle in radians) angle: angle of arrowhead. Default = pi/9
+        :param (class) kwargs: kwargs passed through plotly.graph_objs.Scatter
+            for more information on valid kwargs call
+            help(plotly.graph_objs.Scatter)
+
+        :rtype (trace): returns quiver trace
+
+        :raises (PlotlyError): will happen if x, y, u, and v are not all
+            the same length (or size if ndarray)
+
+        :raises (PlotlyError): will happen if scale or arrow_scale are less
```

Comment:
`ValueError`? You might want to refrain from documenting this information twice, it'd be best to just have it in the `validate` function of `TraceFactory`. DRY holds for documentation too :smiley_cat: